libs/wingc/src/type_check.rs

```diff
@@ -445,6 +442,32 @@ pub trait ClassLike: Display {
 			None
 		}
 	}
+
+	fn phase(&self) -> Phase {
+		// A class is considered a "preflight class" if it has any preflight fields.
+		// Having a preflight state means that different instances of the class
+		// might access different preflight data. Such instances need to be lifted
+		// into inflight individually. Non preflight class instances don't need to be
+		// lifted and can therefore be instantiated inflight.
+		if self
+			.fields(true)
+			.any(|(_, vi)| vi.kind == VariableKind::InstanceMember && vi.phase == Phase::Preflight)
+		{
+			return Phase::Preflight;
+		}
```

Comment:
I'm curious to understand better how come the class phase is determined this way (why it's not determined based on the phase in the parsing phase). Maybe there's an example that would help illustrate this?

Also, could this calculation be done once inside `type_check_class` instead of calculating it each time?

Reply:
I'm scrapping this behavior. See https://winglang.slack.com/archives/C048QCN2XLJ/p1706977736744229
TLDR: bad DX to have the class's phase calculated instead of relying on the `inflight` annotation (or lack thereof).